src/DynamoWebServer/Messages/MessageHandler.cs

```diff
@@ -522,7 +523,27 @@ private string GetInOutPortsData(NodeModel node)
                 stringBuilder.Append("],");
             }
 
-            stringBuilder.Append("\"InPorts\": [");
+            if (node is VariableInputNode)
+            {
+                var type = node.GetType();
+                if (type.Name == "PythonNode")
+                {
+                    var script = type.GetProperty("Script").GetValue(node, null).ToString();
+
+                    stringBuilder.Append("\"Script\":\"");
+                    stringBuilder.Append(script.
+                            Replace("\n", "\\n").
+                            Replace("\"", "\\\""));
+                    stringBuilder.Append("\", ");
+                }
+
+                stringBuilder.Append("\"varInputs\": [");
+            }
+            else
+            {
+                stringBuilder.Append("\"InPorts\": [");
+            }
+
```

Comment:
Hi @pbidenko, while these changes look good, I am not sure if the details can/should be factored out as extensions? We have multiple extensions defined in *DynamoCore\Utilities\Extensions.cs*, you could add an equivalent for `DynamoWebServer/Extensions.cs`, so the code here can look like:

```csharp
var varNode = node as VariableInputNode;
if (varNode != null)
    stringBuilder.Append(varNode.GetInOutPortsData());
```

Then the same can be applied to `CodeBlockNodeModel` too.